lib/MiLight/MiLightRadioPL1197_LT8900.cpp

```diff
@@ -0,0 +1,631 @@
+/**************************************************************************/
+int MiLightRadioPL1167_LT8900::configure()
+{
+  vInitRadioModule(config.type);
+  vSetSyncWord(config.syncword3, 0,0,config.syncword0);
+  vStartListening(config.channels[0]);
+  return 0;
+}
+
+/**************************************************************************/
+// Check if data is available
+/**************************************************************************/
+bool MiLightRadioPL1167_LT8900::available()
+{
+  _waiting = false;
+
+  if(bAvailablePin())
+  {
+    int iStartTime = millis();
+    int iUpdateStamp = 0;
+    static byte byLastReceived[32];
+
+    byte byaFramesSizes[20];
+    byte byaFramesReceivedCount[20];
+    byte byaFramesReceived[20][32];
+    byte byFrameCounter = 0;
+
+    bool bDifference = false;
+
+    bool bRetVal = false;
+
+    if (bAvailablePin())
+    {
+      unsigned long ulTimeStamp = millis();
+      unsigned long ulElapsedTime = 0;
+
+      do
+      {
+        if (bAvailablePin())
+        {
+          iUpdateStamp = millis();
+          uint8_t buf[32];
+
+          int packetSize = iReadRXBuffer(buf, 32);
+          if (packetSize > 0)
+          {
+            bRetVal = true;
+            bDifference = false;
+
+            for (int iCounter = 0; iCounter < packetSize && bDifference == false; iCounter++)
+            {
+              if (buf[iCounter] != byLastReceived[iCounter])
+              {
+                bDifference = true;
+              }
+            }
+
+            if (bDifference == true)
+            {
+              for (int i = 0; i < packetSize; i++)
+              {
+                byaFramesReceived[byFrameCounter][i] = buf[i];
+                byLastReceived[i] = buf[i];
+                byaFramesSizes[i] = packetSize;
+              }
+
+              byaFramesReceivedCount[byFrameCounter] = 1;
+              byFrameCounter++;
+            }
+            else
+            {
+              byaFramesReceivedCount[byFrameCounter-1]++;
+            }
+          }
+          else
+          {
+            if (packetSize < 0)
+            {
+              Serial.println(F("Packet less than zero, buffer to small"));
+            }
+            else
+            {
+              Serial.println(F("Packet read fail"));
+            }
+          }
+
+          vResumeRX();
+        }
+
+        ulElapsedTime = millis();
+        yield();
+        ulElapsedTime = ulElapsedTime - ulTimeStamp;
+      }while (ulElapsedTime < 1000 );
+
+      Serial.print("ElapsedRX: ");
+      Serial.println(iUpdateStamp - iStartTime);
+
+      if (byFrameCounter != 0)
+      {
+        Serial.println("");
+        Serial.print(F("Packets received: "));
+        Serial.print(byFrameCounter);
+
+        for (byte byCounterFrame = 0; byCounterFrame < byFrameCounter; byCounterFrame++)
+        {
+          _dupes_received = byaFramesReceivedCount[byCounterFrame];
+          Serial.println("");
```

Comment:
Can we wrap this and other prints here in `#ifdef DEBUG_PRINTF` or some other toggleable debug flag?